Library/Homebrew/cmd/audit.rb

```diff
@@ -827,12 +827,26 @@ def audit_version
   def audit_checksum
     return unless checksum
 
-    case checksum.hash_type
-    when :md5
-      problem "MD5 checksums are deprecated, please use SHA1 or SHA256"
-      return
-    when :sha1   then len = 40
-    when :sha256 then len = 64
+    strict = ARGV.include? "--strict"
+
+    if strict
+      case checksum.hash_type
+      when :md5
+        problem "MD5 checksums are deprecated, please use SHA256"
+        return
+      when :sha1
+        problem "SHA1 checksums are deprecated, please use SHA256"
+        return
+      when :sha256 then len = 64
+      end
+    else
+      case checksum.hash_type
+      when :md5
+        problem "MD5 checksums are deprecated, please use SHA256"
```

Comment:
@tdsmith If the reason for us to move is for security: I'd say no. Similarly to if they publish `md5` checksums.